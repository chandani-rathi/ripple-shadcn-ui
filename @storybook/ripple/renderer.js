// renderer.js
export async function renderToCanvas({ storyFn, canvasElement }) {
  console.log("renderer")
  const output = storyFn();

  // Example: simple HTML render
  if (typeof output === 'string') {
    canvasElement.innerHTML = output;
  } else if (output instanceof HTMLElement) {
    canvasElement.innerHTML = '';
    canvasElement.appendChild(output);
  } else {
    canvasElement.innerHTML = `<pre>Unsupported output: ${JSON.stringify(output)}</pre>`;
  }
}
