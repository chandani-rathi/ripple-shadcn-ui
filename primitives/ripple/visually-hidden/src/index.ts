export * from './visually-hidden.ripple';