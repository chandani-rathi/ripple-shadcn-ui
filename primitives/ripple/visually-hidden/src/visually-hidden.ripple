import { Primitive } from "@ripple-primitives/primitive-ui"
const NAME = 'VisuallyHidden';

/* -------------------------------------------------------------------------------------------------
 * VisuallyHidden
 * -----------------------------------------------------------------------------------------------*/
const VISUALLY_HIDDEN_STYLES = Object.freeze({
  // See: https://github.com/twbs/bootstrap/blob/main/scss/mixins/_visually-hidden.scss
  position: 'absolute',
  border: 0,
  width: 1,
  height: 1,
  padding: 0,
  margin: -1,
  overflow: 'hidden',
  clip: 'rect(0, 0, 0, 0)',
  whiteSpace: 'nowrap',
  wordWrap: 'normal',
})

export component VisuallyHidden(props){
  <Primitive.span
    {...props}
    style={{ ...VISUALLY_HIDDEN_STYLES, ...props.style }}
  ></Primitive.span>
}