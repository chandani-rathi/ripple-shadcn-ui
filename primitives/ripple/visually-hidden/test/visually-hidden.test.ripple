import { axe } from 'vitest-axe';
import { VisuallyHidden }  from "../src/visually-hidden.ripple";

const TEXT_CHILD = "TEST"

describe("visually-hidden", () => {
    beforeEach(() => {
		component Test(){
			<VisuallyHidden>{TEXT_CHILD}</VisuallyHidden>
		}
		render(Test);
	});

    it('should have no accessibility violations', async () => {
		expect(await axe(container)).toHaveNoViolations();
	});
    
    it("should work", () => {
         expect(container).toMatchSnapshot()
    })
})