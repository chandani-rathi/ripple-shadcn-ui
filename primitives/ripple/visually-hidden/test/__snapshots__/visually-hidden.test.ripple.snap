// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`visually-hidden > should work 1`] = `
<div>
  <!---->
  <!---->
  <!---->
  <span
    style="position: absolute; border: 0px; padding: 0px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
  >
    <!---->
    <!---->
    TEST
    <!---->
    <!---->
    <!---->
  </span>
  <!---->
  <!---->
  <!---->
  
</div>
`;
